### PR TITLE
Update other files

### PR DESCRIPTION
**Pull request by kevinchen93**

### CODE REVIEW

**Comment by coldshower** [comment]
I would say try to name your commits something different. So let's say you finished the name input, that commit can be called "added name input field".

The site looks very polished. Your html structuring is very clean. More comments in the Files Changed Tab.

File: happy-birthday-to-you.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*
```
Comment (coldshower) [inline happy-birthday-to-you.js:1]: This is not as important, but it is good to start early. For JavaScript, convention is to use 2 spaces for indentation. On sublime, you can set syntax specific settings and you can set the indentation to spaces for tabs to be true. Let me know if you have trouble with this. You can continue to use the tab button, but it will just make /s/s rather than /t in the editor. This is good because the code will then look consistent across all machines since a /s is always a /s. But a /t can be different depending on a person's setup.

File: happy-birthday-to-you.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	'Jimmy Feng': 711,
+	'John Chen': 1202,
+	'Kevin Chen': 627,
+	'Kevin Cheung':	521,
+	'Mohammed Rahman': 305,
+	'Robert Tan': 523,
+	'Roland Zhou': 903,
+	'Vainius Glinskis': 805,
+	'Zarif Rahat': 1107,
+
```
Comment (coldshower) [inline happy-birthday-to-you.js]: I like what you are doing here with the object. I feel you should just write the dates a little more normally so it is  just easier to look at.

'5/21' for example. Then to get month, it's just dateString.split('/')[0], since right now you have to do this weird thing where you take everything before the last two digits for the month.

File: happy-birthday-to-you.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+};
+
+function validateInputName() {
+	let text = nameInput.value.toLowerCase();
```
Comment (coldshower) [inline happy-birthday-to-you.js]: lol why did you need to change to lowercase if you check for caps in regex?

File: happy-birthday-to-you.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	// any combination of letters separated by a whitespace
+	let nameFormat = /^[a-zA-Z]+\s+[a-zA-Z]+$/;
```
Comment (coldshower) [inline happy-birthday-to-you.js]: can be const since you don't change it. Basically always use const until you have to use let.

File: happy-birthday-to-you.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	let nameFormat = /^[a-zA-Z]+\s+[a-zA-Z]+$/; 
+
+	return nameFormat.test(text);
+}
+
+function confirmOnList() {
+	let text = nameInput.value.toLowerCase();
+
+	for (let name in birthdayObj) {
+		name = name.toLowerCase();
```
Comment (coldshower) [inline happy-birthday-to-you.js]: I see what you are doing here, but the extra line is probably not needed.

File: happy-birthday-to-you.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+		if (text === name) {
```
Comment (coldshower) [inline happy-birthday-to-you.js]: Can just do `if (text === name.toLowerCase()) {`

File: happy-birthday-to-you.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+			return true;
+		}
+	}
+
+	return false;
+}
```
Comment (coldshower) [inline happy-birthday-to-you.js]: So an alternative would be to leverage the fact that objects can lookup keys really quickly.

To confirm, what you could do is capitalize the name input, and then just see whether the object can look that name up.

For example, 

if I typed `john chen`, you can capitalize that to be `John Chen`, then just do 

`return !!birthdayObj[nameInputValue]`

Let me know if that makes sense.

File: happy-birthday-to-you.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+function getCurrentDate() {
+	const today = new Date();
+	const dd = today.getDate();
+	const mm = today.getMonth() + 1; // January is 0!
+	const yyyy = today.getFullYear();
```
Comment (coldshower) [inline happy-birthday-to-you.js]: not using yyyy so can just remove

File: happy-birthday-to-you.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+goButton.addEventListener('click', function(event) {
+
+	// remove last child of both `<div>`s every time go button is clicked*
+	if (nameInputContainer.lastChild.nodeName === 'H6') {
+		nameInputContainer.lastChild.remove();
+	}
+
+	if (nameInputContainer.lastChild.nodeName === 'H1') {
+		mainContent.lastChild.remove();
+	}
```
Comment (coldshower) [inline happy-birthday-to-you.js]: it's actually quite expensive to remove DOM nodes like this.

A faster alternative would be to simply always have these nodes there, but just let them start as empty. And then you just remove the content or add the content as needed.

File: happy-birthday-to-you.js [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+	// STEP 1 
+	// Ensure that name is in correct format (only strings)
+
+	// variables for validation
+	let isValidatedOrNot = validateInputName();
```
Comment (coldshower) [inline happy-birthday-to-you.js]: Can shorten this to just `isValid`

File: happy-birthday-to-you.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+	const heading6 = document.createElement('h6');
+	heading6.style.margin = '5px';
+
+	// variables for confirmation
+	let text = nameInput.value.toLowerCase();
+	let onListOrNot = confirmOnList();
```
Comment (coldshower) [inline happy-birthday-to-you.js]: Can shorten to just `isOnList`

File: happy-birthday-to-you.js [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+	let todaysDate = getCurrentDate();
+	const heading1 = document.createElement('h1');
```
Comment (coldshower) [inline happy-birthday-to-you.js]: yea so relating to earlier, it might be good to just always have these headings pre-created in html and you just change the content

File: happy-birthday-to-you.js [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+	const birthdayVideo = document.createElement('iframe');
+	
+	// CASE 1: User inputs empty string
+	if (text === '') {
+		heading6.innerHTML = `Please enter a name.`;
+		nameInputContainer.appendChild(heading6);
+		
+	// CASE 2: User inputs a non-empty string but in the incorrect format
+	} else if (isValidatedOrNot === false) {
```
Comment (coldshower) [inline happy-birthday-to-you.js]: equivalent to just `else if (!isValid) {`

File: happy-birthday-to-you.js [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+		heading6.innerHTML = `Please enter your name in the correct format.`;
+		nameInputContainer.appendChild(heading6);
+
+	// STEP 2 
+	// Once correct name format is inputted, compare it to object key/value pairs. 
+
+	// CASE 1: User is a member
+	} else if (onListOrNot === true) {
```
Comment (coldshower) [inline happy-birthday-to-you.js]: can just write `else if (isOnList) {`

File: happy-birthday-to-you.js [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+		for (let name in birthdayObj) {
```
Comment (coldshower) [inline happy-birthday-to-you.js]: ya not sure why we need to iterate again if we already know whether or not they are on list from the `confirmOnList` function

Comment (kevinchen93) [reply happy-birthday-to-you.js]: I needed a way to access the value of each key in the object to do the comparison:  `birthdayObj[name] === getCurrentDate();` on line 113

Comment (coldshower) [reply happy-birthday-to-you.js]: I see! In the case, you may not even need `confirmOnList` since all it does is return `birthdayObj[name]`. You can just save that to a variable and then perform the date check off that!

File: happy-birthday-to-you.js (hunk 13 above)
Comment (kevinchen93) [reply happy-birthday-to-you.js]: Hmm I tried with `!` and it works! I was hesitant to use it since I wasn't quite sure on what behavior to expect and thought it would negate the actual Boolean value of the expression (for example, make it `true` when the name is not a valid name) so I was more comfortable with explicitly making the comparison (similarly with isOnList).
